runtime/compiler/env/VMJ9.cpp

```diff
@@ -3244,6 +3244,7 @@ int TR_J9VMBase::checkInlineableTarget (TR_CallTarget* target, TR_CallSite* call
    // Don't inline methods that are going to be reduced in ilgen or UnsafeFastPath
    switch (rm)
       {
+      case TR::com_ibm_jit_JITHelpers_findElementFromArray:
```

Comment:
so this isn't reduced by ilgen or UnsafeFastPath so the comment above now seems to be misleading...

what happens on platforms that don't support this call in the codegen? I think there needs to be a query so that if this isn't supported in the codegen we add it to the always inline list. The danger being until other codegens support this helper they will see a perf degradation...

Reply:
Good point, I've moved this into the CG specific query `suppressInliningOfRecognizedMethod()`, so that each CG can opt-in/opt-out this optimization.

Reply:
Can we make sure that if the inlining isn't suppressed we put it into the list of always worth inlining to avoid regressions on the platforms which don't initially opt in?

Reply:
In theory the Inliner should make better decision whether to inline `findElementFromArray`, but I see your point about affecting the existing code as little as possible to minimize surprises.
I've update that `findElementFromArray` is always inlined if CG does not request suppressing inlining of it.